src/com/content-event/event-idea.js

```diff
@@ -1,27 +1,26 @@
 import {h, Component} 				from 'preact/preact';
 import NavSpinner						from 'com/nav-spinner/spinner';
-import NavLink 							from 'com/nav-link/link';
 import SVGIcon 							from 'com/svg-icon/icon';
-
-import ButtonBase						from 'com/button-base/base';
-
+import UIButton							from 'com/ui/button/button';
 import $ThemeIdea						from '../../shrub/js/theme/theme_idea';
 
 
 const MAX_IDEAS = 3;
+const canHaveMoreIdeas = (ideas, submitting) => Object.keys(ideas).length + (submitting ? 1 : 0) < MAX_IDEAS;
+const hasSubmittedIdeas = (ideas) => Object.keys(ideas).length > 0;
 
 export default class ContentEventIdea extends Component {
 	constructor( props ) {
 		super(props);
 
 		this.state = {
-			"idea": "",
-			"ideas": null
+			'idea': '',
```

Comment:
This should be:
```js
'idea': "",
```

Another style thing I can't convince eslist to do (I got blocked one time trying to make an eslint plugin that enforced some of my _exotic_ style rules. Alas I didn't have time to really get it working or finished).

* `'` should be used if it's something that will never contain a whitespace/newline (i.e. slugs, keys).
* `"` should be used anywhere there is the potential for whitespace or newlines.

Reply:
This one will be impossible to keep track of ;)

Reply:
It's more about raising errors for certain `'` uses.

To be fair, it might be more work than it's worth. Sticking to `"just double quotes"` might be lesser of evils.